ethportal-peertest/src/lib.rs

```diff
@@ -64,8 +66,7 @@ pub async fn launch_node(id: u16, bootnode_enr: Option<&SszEnr>) -> anyhow::Resu
             TrinConfig::new_from(trin_config_args.iter()).unwrap()
         }
         None => {
-            let ip_addr =
-                socket::find_assigned_ip().expect("Could not find an IP for local connections");
```

Comment:
I think this change isn't necessary, if we change the `--external-address` in tests/self_peertest.rs to use the same IP as the peertest nodes, similar to this derivation:
https://github.com/ethereum/trin/blob/65665e39c4dd831ed0df5f4cd5f753aaf1b9febb/ethportal-peertest/src/lib.rs#L43-L47

If it *is* necessary, then there are some other cleanups we should do, including this ^ code.

Reply:
There are some things that concern me:

1. `find_assigned_ip` is not stable, it is not returning the same IP interface on every run. What is the point then of using this method if it is returning random results? I'm more in favor to get rid of this method and adding helper method that always returns `127.0.0.1` for example.

2.  Running `find_assigned_ip` just to get some random IP for one node and then use the same IP for the next node is kinda confusing, at least for me.
I think it is less mentally overwhelming to just run every node with a fixed static address which is not a result of a random method.

Reply:
Yeah, I think I used to have a reason I didn't like using 127.0.0.1 here (maybe I wanted to be able to connect to non-local clients?). But I can't seem to defend it now, so I'm good with using localhost. As more tests get added, it's clearer that the complexity cost of picking a constant IP is growing...

There's probably some more work that can be done to simplify, and use 127.0.0.1 everywhere in the tests.